### ISSUE TITLE
Rounding issues when formatting with Degree_Minutes_Seconds

### ISSUE THREAD

**Issue by B4rrCH** [Tronald/CoordinateSharp#170]
**Describe the bug**
When formatting the Coordinates with degrees, minutes and seconds, it can sometimes happen, that we get strings of the form `ddd° mm' 60"`. This causes issues when trying to serialize and later de-serialize coordinates, as the parser is not happy with this.

**To Reproduce**
```csharp
Coordinate c1 = Coordinate.Parse("18 19 54 44 59 0");
// c1.Longitude.Minutes = 58
// c1.Longitude.Seconds = 59.99999999988
CoordinateFormatOptions options = new CoordinateFormatOptions
{
    Format = CoordinateFormatType.Degree_Minutes_Seconds
};
string c1_string = c1.ToString(options);
// c1_string = "N 18º 19' 54\" E 44º 58' 60\""
/*
    Put c1_string into Json/XML/csv, ...
    Then later:
*/
Coordinate c2 = Coordinate.Parse(c1_string);
// Throws System.FormatException: "Input Coordinate \"N 18º 19' 54\" E 44º 58' 60\"\" was not in a correct format."
```

**Expected behavior**
I'd expect `c1_string` to be `N 18º 19' 54" E 44º 59' 00"`, so we can de-serialize the output of ToString without having to deal with edge cases as user.

**Environment:**
 - .NET Framework 4.7.2
 - CoordinateSharp Version 2.8.1.1,  Runtime v4.0.30319.
 - Found in randomized Microsoft.VisualStudio.TestTools.UnitTesting-tests for a Windows Forms app. 
 - No multithreading in the unit tests

**Additional Info**
Found when I ran randomized Tests where I serialize and then de-serialize objects with Coordinate properties and compare them.

**Comment by Tronald** [comment]
Wow, good find. I can't believe this issue hasn't come up yet.

It appears to be a float point precision loss occurring due to conversion combined with the string handler not knowing what to do with it.

Just to kind of explain. Coordinates are stored as doubles for maximum precision, BUT they are converted as decimals to minimize precision loss during conversions. The parsers however work in doubles to allow maximum precision to be fed in, so you can experience slight precision loss during a parse (less than a mm which is well within deltas for purposes of this library).

44 59 0 is converting to 44 58 59.99999999988

The string handler is rounding to 60 (which is not good), because it thinks that's what it's supposed to do. I will work on getting this fixed before the next release come out.

Thank you for reporting this.

**Comment by Tronald** [comment]
Correction implemented in `ToString()` logic. Will push with next update (hopefully in a couple of weeks). Our build is done, but we still have to implement tests and documentation. If you need a quick fix, I can push an 2.9.1.1 Alpha version out for you just let me know.

**Comment by B4rrCH** [comment]
Oh wow, that was fast. It's not very urgent, but thanks.

**Comment by Tronald** [comment]
Fixed with v2.9.1.3 published on 11/29/2020.
